### PR TITLE
ci: Rename Linux jobs correctly.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
The shared job should not disable pkg-config usage and should be the one that does a Harfbuzz build.

The static job should disable pkg-config and should not do its own Harfbuzz build outside of the `harfbuzz-sys` build scripts.

This is a precursor to making sure things are actually set up correctly and clearly doing the appropriate thing.

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
@mrobinson This is the start of my take on things ...

There are several confusing things going on.

1) The build script would silently fall back from one behavior to the other.
2) One of the behaviors could be disabled via an environment variable rather than a feature.
3) The macOS build wasn't testing the vendored / bundled library.
4) The Linux build is doing a build of Harfbuzz in a difficult way when it could (perhaps) use a version from the package manager. Using the version from the package manager is either good or bad in that it forces us to support multiple versions of harfbuzz.
5) The default behavior of whether it built or used pkg-config wasn't clearly indicated anywhere.
6) I didn't even look at what is happening with `freetype`.

In my changes here, I try a couple of things in layers, so that we can pull in any subset of them:

1) Rename the Linux jobs to be clear about which is doing what.
2) Add a macOS job that improves build configuration coverage.
3) Change the build script so that it uses the vendored / bundled harfbuzz by default and won't check `pkg-config`. If `--features use-pkg-config-harfbuzz` is specified, then it will try to use `pkg-config` to locate Harfbuzz and fail otherwise, not fall back to the vendored library.
4) Try using the system harfbuzz on Linux. I don't necessarily agree with this.

I'm open to renaming features or seeing what other comments you have ...

If we land items 1-3, we can either land 4 or change how we build harfbuzz to use meson, and then we can switch to the submodule to pull in the vendored sources.

Thoughts?

File: harfbuzz-sys/Cargo.toml [hunk 1]
```diff
@@ -46,5 +46,6 @@ optional = true
 
 [features]
 default = ["build-native-harfbuzz", "build-native-freetype"]
-build-native-harfbuzz = ["cc", "pkg-config"]
+build-native-harfbuzz = ["cc"]
 build-native-freetype = ["freetype", "freetype/freetype-sys"]
+use-pkg-config-harfbuzz = ["pkg-config"]
```
Comment (mrobinson) [inline harfbuzz-sys/Cargo.toml:51]: There are a couple issues here:

1. This changes  the default behavior of the crate. Before it would try to use the system version of HarfBuzz unless you set an environment variable. I don't think we can do this without bumping the major version number.  I think the default behavior should follow what most `-sys`-type crates do, whatever that is.
2. This feature shouldn't refer to `pkg-config`, because `pkg-config` is an implementation detail of how you find the system shared library. The most important thing here is that a feature like this controls whether or not we are dynamically linking to the system library or including a statically compiled version of HarfBuzz. That's why I chose the `static` name when I tried to make this configurable as a feature. We could discuss a better name, but maybe this change could be written on top of mine?

File: harfbuzz-sys/build.rs [hunk 2]
```diff
@@ -1,17 +1,15 @@
+#[cfg(feature = "use-pkg-config-harfbuzz")]
+fn find_harfbuzz_via_pkgconfig() {
+    pkg_config::probe_library("harfbuzz").is_ok();
```
Comment (mrobinson) [inline harfbuzz-sys/build.rs:3]: This function just calls a single function. I think it could just be inline.

File: harfbuzz-sys/build.rs [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
+}
+
 #[cfg(feature = "build-native-harfbuzz")]
-fn main() {
+fn build_harfbuzz() {
```
Comment (mrobinson) [inline harfbuzz-sys/build.rs:7]: What is the meaning of `native` here? I understand `native` to mean the version of HarfBuzz that comes with the system...but it's very confusing so maybe we should rename it.

File: harfbuzz/Cargo.toml [hunk 4]
```diff
@@ -20,3 +20,4 @@ default-features = false
 default = ["build-native-harfbuzz", "build-native-freetype"]
 build-native-harfbuzz = ["harfbuzz-sys/build-native-harfbuzz"]
 build-native-freetype = ["harfbuzz-sys/build-native-freetype"]
+use-pkg-config-harfbuzz = ["harfbuzz-sys/use-pkg-config-harfbuzz"]
```
Comment (mrobinson) [inline harfbuzz/Cargo.toml:23]: What does it mean when `build-native-freetype` and `build-native-harfbuzz` aren't active?

File: harfbuzz-sys/makefile.touch [hunk 5]
```diff
@@ -1,45 +0,0 @@
-# Check that $(HARFBUZZ) is defined before trying to use it.
```
Comment (mrobinson) [inline harfbuzz-sys/makefile.touch:1]: Is the removal of this file related to the rest of the change or could it be a separate PR?

**Review by mrobinson — CHANGES_REQUESTED** [review]
I do really like the idea of making the CI jobs clearer and accurately named, but I think there is some more discussion about some other aspects of this change. Happy to chat if that's easier.

File: harfbuzz-sys/build.rs (hunk 3 above)
Comment (waywardmonkeys) [reply harfbuzz-sys/build.rs:7]: No idea anymore. It is confusing and I'd rather remove all of it really.

File: harfbuzz/Cargo.toml (hunk 4 above)
Comment (waywardmonkeys) [reply harfbuzz/Cargo.toml:23]: Not sure about the `build-native-freetype`, but currently, if you `cargo test --no-default-features`, it doesn't link against any `harfbuzz` libs and so it fails to link with undefined symbols.

File: harfbuzz-sys/makefile.touch (hunk 5 above)
Comment (waywardmonkeys) [reply harfbuzz-sys/makefile.touch:1]: It is only used by that code in the build job that is building harfbuzz on Linux, so once that code disappeared in a commit, this went with it.

File: harfbuzz-sys/build.rs (hunk 2 above)
Comment (waywardmonkeys) [reply harfbuzz-sys/build.rs:3]: True. My first version had more here than just this and I didn't think to revise it when I eliminated the other code.

**Comment by waywardmonkeys** [comment]
Once #217 (hopefully) lands, I'll rebase this and see what the next step is.
